packages/next/src/export/routes/app-route.ts

```diff
@@ -141,6 +142,29 @@ export async function exportAppRoute(
       'utf8'
     )
 
+    const routeName = normalizedPage.split('/').pop()
```

Comment:
The assertion should be moved into `packages/next/src/build/webpack/loaders/next-metadata-route-loader.ts` `getStaticAssetRouteCode` function, `routeName` is not realiable for all cases.

Reply:
Thanks!